code-challenges.test.js

```diff
@@ -1,59 +1,168 @@
+
+// // a) Create a test with expect statements for each of the variables provided.
+
+// describe("toUpperCases", () => {
+//     it("returns the array caplitalized", () => {
+//       expect(toUpperCases(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+//       expect(toUpperCases(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+//     })
+//   })
+
+
+// // b) Create the function that makes the test pass.
+
+// const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+// // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+// const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+// // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+
+
+
+// // //PSEUDO Code
+// // //make a function that takes in an array of words and returns an array with all the words capitalized..
+// // //input: randomNuouns 1-2
+// // //output:  randomNuouns 1-2
+// // declare a function that takes in an array
+// // access the first value of each index and cappitalize it , and then concat the rest of the values to make uppercase and the other letter to be lowercase
+// // logging the function and pass in the array
+
+// //
+// const toUpperCases = (array) => {  return array.map((value) => {return value[0].toUpperCase() + value.substring(1)  })}
+
+
+    
+    
+
+
+
+
+
+
+
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
+
+
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
 
-// b) Create the function that makes the test pass.
+// describe("vowelFinder", () => {
+//     it("logs the index of the first vowel.", () => {
+//       // expect(vowelFinder(vowelTester1)).toEqual(1)     
+//       // expect(vowelFinder(vowelTester2)).toEqual(0)
+//       //  expect(vowelFinder(vowelTester3)).toEqual(2)
+//     })
+//   })
+
+
+// const vowelTester1 = "learn"
+// // Expected output: 1
+// const vowelTester2 = "academy"
+// // Expected output: 0
+// const vowelTester3 = "challenges"
+// // Expected output: 2
+
+// // b) Create the function that makes the test pass.
+
+//  //PSEUDO Code
+// // make a function that takes in a string and logs the index of the first vowel.
+// //input: voweltester 1-3
+// //output: index of the first vowel of teh string
+// // declare a function that takes in an string
+// // access the first vowel of each string and return the index number 
+
+// // Student comments : i cant seem to get the function to be dynamic where it can search for mulitpy letters
+// // comment:you have to run each one seperate inorder to make it work
+
```

Comment:
One of the approaches we can take here is to create an array with vowels. Then, in our function we can find the index of the first vowel we come across. I would give this one another attempt!